cmd/trcd/main.go

```diff
@@ -29,31 +33,89 @@ func init() {
 func main() {
 	flag.Parse()
 
-	conn, err := net.Listen("unix", *sock)
+	var netConn net.Listener
+	var err error
+	if *useTCP {
+		log.Debug("Listening to TCP socket...")
+		addr, _ := net.ResolveTCPAddr("", ":4243")
+		netConn, err = net.ListenTCP("tcp", addr)
+
+	} else {
+		log.Debug("Listening to Unix socket...")
+		netConn, err = net.Listen("unix", *sock)
+	}
 	if err != nil {
 		panic(err)
 	}
-	defer conn.Close()
+	defer netConn.Close()
 
 	for {
-		c, err := conn.Accept()
+		sockConn, err := netConn.Accept()
+		log.Infof("Connected to %s", sockConn.RemoteAddr())
 		if err != nil {
 			panic(err)
 		}
-		defer c.Close()
+		defer sockConn.Close()
 
+		trcConn := Connect(sockConn, sockConn,
+			WithHandler(api.MessageTypeState, setStateHandler),
+			WithHandler(api.MessageTypePing, pingHandler),
+		)
+
+		// print errors resulting from mock trc
 		go func() {
-			dec := json.NewDecoder(c)
-			dec.DisallowUnknownFields()
-			var msg api.Message
-			if err := dec.Decode(&msg); err != nil {
-				log.Printf("Failed to decode input: %s", err)
+			for err := range trcConn.Errors() {
+				log.Error(err)
 			}
-			log.Printf("Got: %v", msg)
 		}()
-		if _, err := io.Copy(c, os.Stdin); err != nil {
-			log.Printf("Failed to copy input: %s", err)
+
+		// mock trc is running parallel to client anyway
```

Comment:
what's the purpose of this comment?

Reply:
as answer to why it is not running in a go routine